tcbuilder/cli/build.py

```diff
@@ -438,24 +444,85 @@ def handle_bundle_output(image_dir, storage_dir, bundle_props, tezi_props):
                 "use_host_docker": False,
                 "output_filename": common.DOCKER_BUNDLE_FILENAME,
                 "keep_double_dollar_sign": bundle_props.get("keep-double-dollar-sign", False),
-                "platform": platform
+                "platform": platform,
+                "compress": is_compressed_tar
             }
             download_containers_by_compose_file(**download_params)
 
-            # Do a combine "in place" to avoid creating another directory.
-            combine_params = {
-                "image_dir": image_dir,
-                "bundle_dir": bundle_dir,
-                "output_directory": None,
-                "tezi_props": translate_tezi_props(tezi_props),
-                "force": True
-            }
-            comb_be.combine_tezi_image(**combine_params)
+            return bundle_dir
 
-        finally:
+        except:
+            raise TorizonCoreBuilderError(
+                "Error trying to bundle Docker containers"
+            )
+
+    return None
+
+
+def handle_bundle_output(image_dir, storage_dir, bundle_props, tezi_props):
+    """Handle the bundle and combine steps of the output generation."""
+
+    try:
+        bundle_dir = handle_bundle_common(
+            storage_dir,
+            bundle_props
+        )
+
+        if bundle_dir is None:
+            return
+
+        # Do a combine "in place" to avoid creating another directory.
+        combine_params = {
+            "image_dir": image_dir,
+            "bundle_dir": bundle_dir,
+            "output_directory": None,
+            "tezi_props": translate_tezi_props(tezi_props),
+            "force": True
+        }
+        comb_be.combine_tezi_image(**combine_params)
+
+    finally:
+        if bundle_dir is not None and os.path.exists(bundle_dir):
+            log.debug(f"Removing temporary bundle directory {bundle_dir}")
+            shutil.rmtree(bundle_dir)
+
+
+def handle_raw_image_bundle_output(
+        image_dir,
+        raw_image_path,
+        storage_dir,
+        bundle_props,
+        raw_props
+    ):
```

Comment:
nit pick: odd indentation (why not follow the same pattern as `handle_bundle_output()`?)

Reply:
Should be solved.